### PR TITLE
Add an example Nushell plugin written in Nushell itself

### PR DESCRIPTION
**Pull request by devyn**
# Description

As suggested by @fdncred.

It's neat that this is possible, but the particularly useful part of this is that we can actually
test it because it doesn't have any external dependencies, unlike the python plugin.

Right now this just implements exactly the same behavior as the python plugin, but we could have it
exercise a few more things.

Also fixes a couple of bugs:

- `.nu` plugins were not run with `nu --stdin`, so they couldn't take input.
- `register` couldn't be called if `--no-config-file` was set, because it would error on trying to
  update the plugin file.

# User-Facing Changes

- `nu_plugin_nu_example` plugin added.
- `register` now works in `--no-config-file` mode.

# Tests + Formatting
Tests added for `nu_plugin_nu_example`.

- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :green_circle: `toolkit test`
- :green_circle: `toolkit test stdlib`

# After Submitting

- [ ] Add the version bump to the release script just like for python

### CODE REVIEW

File: crates/nu_plugin_nu_example/nu_plugin_nu_example.nu [hunk 1]
```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env -S nu --stdin
```
Comment (fdncred) [inline crates/nu_plugin_nu_example/nu_plugin_nu_example.nu:1]: A shebang makes me wonder about this working with Windows.

Comment (fdncred) [reply crates/nu_plugin_nu_example/nu_plugin_nu_example.nu:1]: I guess it works since the CI is green. yeehaw!

Comment (devyn) [reply crates/nu_plugin_nu_example/nu_plugin_nu_example.nu:1]: Handled specifically by the plugin case along with other extensions. The shebang actually doesn't matter

**Comment by fdncred** [comment]
This is very creative. Super nice work. I always wanted to have plugins written in nushell. 

After going through it a bit, it makes me wonder if, in the future, we could have a module that would abstract some of this and make building plugins in nushell super trivial. 

I think about `jc` and how it wraps externals and think we could do something similar with script plugins. Not that I want to reinvent jc, but it would be cool to have something that was pure nushell that wrapped externals that people could contribute to.

**Comment by devyn** [comment]
I have to wonder though, is there really any advantage to doing this over a script library?

There's a lot more to have to handle to be able to do asynchronous streams, and I'm not totally sure how possible it is.

If it's Nushell anyway why not just wrap all of the external commands with parsers and expose custom commands in a module?

**Comment by fdncred** [comment]
Yup, I know what you're saying. Regardless of whether it's true or not, I just tend to think about plugins as being "deeply" integrated into nushell and scripts being "just scripts". LOL. 

I also think about nupm downloading script plugins. You make me think about it more though. Are script plugins just an easy way for other scripting languages to create nushell plugins?

I wonder if there are any advantages or perhaps disadvantages to having a script plugin?

I still think this PR is way cool.

**Comment by devyn** [comment]
I think it could be useful to do this if there's a library in a scripting language that you really want to use. But probably never any point in Nushell - we found probably the only use case, to allow us to have something to test without external deps 😊 

I think modules are just as good as plugins. Better, maybe, because they have some things like auto-completion signatures that plugins don't have right now. And as nupm matures I think there really shouldn't be much difference between using a plugin library and using a module library. You're just installing it either way and it doesn't matter how it's written.

This was fun though.

**Comment by kubouch** [comment]
I also think that there is no reason to use plugins in Nushell as you can use Nushell directly. One important exception is FUN, of course.

@devyn Can you also update the [version bump script](https://github.com/nushell/nu_scripts/blob/main/make_release/bump-version.nu) to cover this?

**Comment by devyn** [comment]
Will do.

**Pull request merged**